lib/opam_build.ml

```diff
@@ -95,15 +100,25 @@ let install_project_deps ~opam_version ~opam_files ~selection =
   @ pin_opam_files ~network groups @ [
     env "DEPS" non_root_pkgs;
     opam_depext;
+    unpin_opam_files ~network opam_files;
     run ~network ~cache "opam install $DEPS" ]
+  @ pin_opam_files ~network groups
+  @ [ run  ~network ~cache "opam install %s" root_pkgs ]
 
 let spec ~base ~opam_version ~opam_files ~selection =
   let open Obuilder_spec in
+  let build_each_package =
+    List.map
+        (fun x ->
+            Filename.basename x
+            |> Filename.chop_extension
+            |> run "opam exec -- dune build --only-packages=%s @install @check @runtest && rm -rf _build")
+        opam_files
+  in
   stage ~from:base (
     comment "%s" (Fmt.str "%a" Variant.pp selection.Selection.variant) ::
     user ~uid:1000 ~gid:1000 ::
-    install_project_deps ~opam_version ~opam_files ~selection @ [
-      copy ["."] ~dst:"/src/";
-      run "opam exec -- dune build @install @check @runtest && rm -rf _build"
-    ]
+    copy ["."] ~dst:"/src/" ::
```

Comment:
Doing this before deps installation is going to hurt caching, I think? ocaml-ci's assumption is that dependencies change less frequently, so it's trying to install dependencies first so that _different PRs_ which change code, but not the dependency cone, share the deps.